source/agora/test/Base.d

```diff
@@ -242,7 +242,7 @@ public class TestAPIManager
     public void createNewNode (PublicKey address, Config conf)
     {
         auto api = RemoteAPI!TestAPI.spawn!(TestNode)(conf, &this.reg);
-        this.reg.register(address.toString(), api.tid());
+        enforce(this.reg.register(address.toString(), api.tid()));
```

Comment:
Actually this commit is a bit misguided. The `spawn` call there is asynchronous, so we cannot guarantee the child thread has finished doing its work by the time this enforce is called. Classic data race.

It's still good to keep the check here, but I will change it to an assert, and also change the commit description.

Reply:
Resolved.

Reply:
Not resolved, this commit needs to be removed. As mentioned, the mbox can become closed if  the thread shuts down due to a faulty constructor. So this assert/enforcement will naturally fail.

Reply:
Now it's removed.